src/step_db.cpp

```diff
@@ -62,6 +62,7 @@ bool fields_chunk::step_db(field_type ft) {
         realnum *f_p = have_p ? f[c_p][cmp] : NULL;
         realnum *f_m = have_m ? f[c_m][cmp] : NULL;
         realnum *the_f = f[cc][cmp];
+        bool need_bfast = (bfast_scaled_k[0] || bfast_scaled_k[1] || bfast_scaled_k[2]) ? 1 : 0;
```

Comment:
The result of `||` should already be a boolean value (`1` or `0`).
```suggestion
        bool need_bfast = bfast_scaled_k[0] || bfast_scaled_k[1] || bfast_scaled_k[2];
```

Reply:
Fixed.